packages/prisma/migrations/20251003103832_upsert_watchlist_audit/migration.sql

```diff
@@ -0,0 +1,113 @@
+-- CreateEnum
+CREATE TYPE "WatchlistSource" AS ENUM ('MANUAL', 'FREE_DOMAIN_POLICY');
+
+-- AlterEnum
+ALTER TYPE "WatchlistAction" ADD VALUE 'ALERT';
+
+
+-- 1) Add a temporary UUID column `uid` and backfill it without extensions
+ALTER TABLE "Watchlist"
+  ADD COLUMN IF NOT EXISTS "uid" UUID;
+
+-- Backfill existing rows with UUIDs using gen_random_uuid()
+UPDATE "Watchlist"
+SET "uid" = COALESCE("uid", gen_random_uuid());
+
+-- Enforce NOT NULL now that all rows have a value
+ALTER TABLE "Watchlist"
+  ALTER COLUMN "uid" SET NOT NULL;
+
+-- 2) Adjust uniques/indexes that reference old id? (none in your snippet)
+--    If you had FKs or PK references elsewhere, you'd update them here.
+
+-- 3) Switch primary key from old `id` to new `uid`:
+--    Drop old PK (name might vary in your DB; adjust if needed)
+DO $$ BEGIN
+  ALTER TABLE "Watchlist" DROP CONSTRAINT "Watchlist_pkey";
+EXCEPTION
+  WHEN undefined_object THEN NULL;
+END $$;
+
+-- 4) Drop the old `id` column and rename `uid`→`id`
+ALTER TABLE "Watchlist" DROP COLUMN "id";
+ALTER TABLE "Watchlist" RENAME COLUMN "uid" TO "id";
+
+-- 5) Add the new PK on `id` (UUID)
+ALTER TABLE "Watchlist" ADD CONSTRAINT "Watchlist_pkey" PRIMARY KEY ("id");
+
+-- Create WatchlistAudit table BEFORE modifying Watchlist
+-- CreateTable
+CREATE TABLE "WatchlistAudit" (
+    "id" UUID NOT NULL,
+    "type" "WatchlistType" NOT NULL,
+    "value" TEXT NOT NULL,
+    "description" TEXT,
+    "action" "WatchlistAction" NOT NULL DEFAULT 'REPORT',
+    "changedAt" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    "changedByUserId" INTEGER,
+    "watchlistId" UUID NOT NULL,
+
+    CONSTRAINT "WatchlistAudit_pkey" PRIMARY KEY ("id")
+);
+
+-- DropForeignKey
+ALTER TABLE "Watchlist" DROP CONSTRAINT "Watchlist_createdById_fkey";
+
+-- DropForeignKey
+ALTER TABLE "Watchlist" DROP CONSTRAINT "Watchlist_updatedById_fkey";
+
+-- Modify Watchlist table structure
+-- AlterTable
+ALTER TABLE "Watchlist" DROP COLUMN "createdAt",
+DROP COLUMN "createdById",
+DROP COLUMN "severity",
+DROP COLUMN "updatedAt",
+DROP COLUMN "updatedById",
+ADD COLUMN     "isGlobal" BOOLEAN NOT NULL DEFAULT false,
+ADD COLUMN     "lastUpdatedAt" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+ADD COLUMN     "source" "WatchlistSource" NOT NULL DEFAULT 'MANUAL';
```

Comment:
We are currently not backfilling WatchlistAudit table, however, I wonder if we need to backfill it, since althought it's empty right now (new table), but it would contain data from the Watchlist table that we are dropping (and would be keeping in the new WatchlistAudit table intead). For example, `Watchlist.createdBy` would no longer exist, but `watchlistAudit.changedByUserId` does. Another example is, `Watchlist.udpatedAt` -> `WatchlistAudit.changedAt`. What is our move?

The challenge with backfilling here is:
PostgreSQL doesn't have built-in UUIDv7 support, so we'd either need to break this PR down into two parts (first table creation, then a script to backfill and then dropping the columns in a follow up)
Or use an extension such as pg_uuidv7

Reply:
Skipping for now after consulting with @emrysal